lta/transfer/service.py

```diff
@@ -0,0 +1,39 @@
+# service.py
+"""Module providing a TransferService base class."""
+
+import copy
+import importlib
+from typing import Any, Dict, Union
+
+from ..lta_types import BundleType
+
+TransferReference = str
+TransferServiceConfig = Dict[str, Any]
+TransferSpec = BundleType
+TransferStatus = Dict[str, Union[bool, str]]
+
+class TransferService:
+    """TransferService is an abstract base class that specifies an interface to transfer services."""
+
+    def __init__(self, config: TransferServiceConfig):
+        """Initialize a TransferService object."""
+        self.config = copy.deepcopy(config)
+
+    async def start(self, spec: TransferSpec) -> TransferReference:
+        """Ask the TransferService to start the specified transfer."""
+        raise NotImplementedError("TransferService.execute() is abstract and must be implemented in a subclass")
```

Comment:
I assume the function name changed.

Reply:
Good catch. Yes, originally I thought `execute()` but then decided that `start()` more accurately described what the user was actually doing.